publishers/HumanStatePublisher/HumanStatePublisher.cpp

```diff
@@ -90,6 +92,36 @@ bool HumanStatePublisher::open(yarp::os::Searchable& config)
         useDefaultPeriod = true;
     }
 
+    yarp::os::Bottle* fixedBasePosition;
+    if (config.check("fixBasePosition")) {
```

Comment:
@lrapetti having the nested if conditions such as here looks cumbersome. Try to clean this section to have minimum number  of conditional logic. Usually it is clean to have a parameter check first and then parsing the parameters at a later stage during the initialization. For reference please check https://github.com/robotology/human-dynamics-estimation/blob/40e949a4e09aad4a412b0f7af107d80d7831ce7b/devices/HumanStateProvider/HumanStateProvider.cpp#L218 and https://github.com/robotology/human-dynamics-estimation/blob/40e949a4e09aad4a412b0f7af107d80d7831ce7b/devices/HumanStateProvider/HumanStateProvider.cpp#L304

Reply:
The reason for this structure is that the `fixBasePosition` option is an optional flag.
So, if the configuration is present we check if its value is valid, otherwise, we simply skip the option (this because most of the time this option is not used).

Reply:
Makes sense in the case of optional parameter.